### DISCUSSION TITLE
System Encoding Value - default in containers is "ANSI_X3.4-1968" ?

### DISCUSSION THREAD

**Discussion by jsorah** [keycloak/keycloak#11527] [Q&A] [answered]
After running Keycloak in containers, we noticed that the System Encoding is set to `ANSI_X3.4-1968` - however, if I were to run it locally using the source code and not a container, I see UTF-8.

The way I understand this can come about is it is derived from the system locale (on RHEL based systems there's a script that is often sourced - `/etc/profile.d/lang.sh` ) and the `ANSI_X3.4-1968` value can happen when there isn't a locale set (default).

I am curious as to what impacts this could have
- Are there possibly edge cases or errors that could come up? 
- Since the `ANSI_X3.4-1968` charset is basically ASCII, that could mean there are potentially issues for localization?
- Is it possible that this value is ignored and doesn't matter based on how Keycloak works?
- If I update my container to actually set UTF-8, would one expect problems due to the switch?
- Is there some instruction I am missing that we _should_ be setting the `file.encoding` Java system property manually?

EDIT: 2022-04-20

Looking closer, it appears this was actually handled in the "legacy" Keycloak image (Wildfly distribution) - https://github.com/keycloak/keycloak-containers/blob/main/server/Dockerfile#L12

Running the legacy image I was able to confirm that UTF-8 is reflected on the Server Info page.

`docker run -p 8080:8080 -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -it quay.io/keycloak/keycloak:17.0.1-legacy`

```
User Locale 	us_EN
System Encoding 	UTF-8
```

I do not see any of the similar fixes applied in the quarkus image build:
https://github.com/keycloak/keycloak/blob/main/quarkus/container/Dockerfile

And running that image I could confirm the `ANSI_X3.4-1968` value is set.

`docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:17.0.1 start-dev`

```
User Locale 	us_EN
System Encoding 	ANSI_X3.4-1968
```

And while not purely a Keycloak concern, it seems that RHSSO image is in similar shape, where the locale value is using the default - https://github.com/jboss-container-images/redhat-sso-7-openshift-image/blob/sso75-dev/image.yaml

`docker run -p 8080:8080 -e SSO_ADMIN_USERNAME="admin" -e SSO_ADMIN_PASSWORD="admin" -it registry.redhat.io/rh-sso-7/sso75-openshift-rhel8`

```
User Locale 	us_EN
System Encoding 	ANSI_X3.4-1968
```

**Comment by DGuhr** [comment, marked as answer]
To wrap up: Fixed in #11545 via pr #11547
